### PR TITLE
[DE-1079] Events tests

### PR DESCRIPTION
**Pull request by maciej-nedza**
please disable these tests if they start to flak out because of delays in events arrival

### CODE REVIEW

File: tests/src/test/java/com/maxio/advancedbilling/utils/matchers/EventSpecificDataGetter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import com.maxio.advancedbilling.models.ProformaInvoiceIssued;
+import com.maxio.advancedbilling.models.RefundSuccess;
+import com.maxio.advancedbilling.models.SubscriptionGroupSignupEventData;
+import com.maxio.advancedbilling.models.SubscriptionProductChange;
+import com.maxio.advancedbilling.models.SubscriptionStateChange;
+import com.maxio.advancedbilling.models.containers.EventEventSpecificData;
+
+public class EventSpecificDataGetter<R> implements EventEventSpecificData.Cases<R> {
+
+
```
Comment (patryk-grudzien-keen) [inline tests/src/test/java/com/maxio/advancedbilling/utils/matchers/EventSpecificDataGetter.java:25]: Minor - extra blank line(s)

File: tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+    Customer customer;
+    Subscription subscription;
+    Customer updatedCustomer;
+    String customerFullName;
+    String updatedCustomerFullName;
+    long lastEventId = 1;
+
+    @BeforeAll
+    void setup() throws IOException, ApiException, InterruptedException {
+        Thread.sleep(3000);
```
Comment (patryk-grudzien-keen) [inline tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java:64]: Why sleep at the beginning?

File: tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+import static com.maxio.advancedbilling.models.EventKey.*;
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class EventsControllerListEventsTest {
+
+    private final AdvancedBillingClient client = TestClientProvider.getClient();
+    private final TestSetup testSetup = new TestSetup();
+
+    ProductFamily productFamily;
```
Comment (patryk-grudzien-keen) [inline tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java:51]: Minor - make these fields private.

File: tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+    void shouldListEvents() throws IOException, ApiException {
+        // given - when
+        List<EventResponse> events = client.getEventsController().listEvents(new ListEventsInput.Builder()
+                .sinceId(lastEventId)
+                .perPage(200)
+                .direction(Direction.DESC)
+                .build()
+        );
+        assertThat(events.size()).isBetween(13, 14);
+        if (events.size() > 13) {
```
Comment (patryk-grudzien-keen) [inline tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java:136]: This `if` is not needed.

File: tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+
+        // then
+        for (EventResponse event : events) {
+            assertCommonFields(event.getEvent());
+        }
+
+        assertThat(eventsMap.keySet()).containsExactlyInAnyOrder(PAYMENT_SUCCESS, METERED_USAGE);
+
+        Event event = eventsMap.get(PAYMENT_SUCCESS);
+        assertThat(event.getMessage()).isEqualTo("Successful payment of $10.00 for %s's subscription to %s"
```
Comment (patryk-grudzien-keen) [inline tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java:288]: Assertions for `PAYMENT_SUCCESS` and `METERED_USAGE` could be extracted and reused.

**Review by patryk-grudzien-keen — APPROVED** [review]
LGTM 👍 Added some suggestions.

File: tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java (hunk 2 above)
Comment (maciej-nedza) [reply tests/src/test/java/com/maxio/advancedbilling/controllers/events/EventsControllerListEventsTest.java:64]: sometimes events are appearing with a little delay. Without it, events caused by execution of other tests could be still arriving during setup

**Review by patryk-grudzien-keen — APPROVED** [review]

**Pull request merged**